src/storage/src/source/mysql/replication/events.rs

```diff
@@ -42,33 +42,46 @@ fn table_ident(name: &str, current_schema: &str) -> Result<MySqlTableName, Trans
 /// the database, unless the query is logged row-based.' This means that we can
 /// expect any DDL changes to be represented as QueryEvents, which we must parse
 /// to figure out if any of the tables we care about have been affected.
+///
+/// This function returns a bool to represent whether the event that was handled
+/// represents a 'complete' event that should cause the frontier to advance beyond
+/// the current GTID.
 pub(super) async fn handle_query_event(
     event: QueryEvent<'_>,
     ctx: &mut ReplContext<'_>,
     new_gtid: &GtidPartition,
-) -> Result<(), TransientError> {
+) -> Result<bool, TransientError> {
     let (id, worker_id) = (ctx.config.id, ctx.config.worker_id);
 
     let query = event.query();
     let current_schema = event.schema();
+    let mut is_complete_event = false;
 
     // MySQL does not permit transactional DDL, so luckily we don't need to
     // worry about tracking BEGIN/COMMIT query events. We only need to look
     // for DDL changes that affect the schema of the tables we care about.
     let mut query_iter = query.split_ascii_whitespace();
     let first = query_iter.next();
     let second = query_iter.next();
-    if let (Some(first), Some(second)) = (first, second) {
+    match (
+        first
+            .map(str::to_ascii_lowercase)
+            .as_ref()
+            .map(String::as_str),
+        second
+            .map(str::to_ascii_lowercase)
+            .as_ref()
+            .map(String::as_str),
```

Comment:
rust tip: `.as_ref().map(String::as_str)` is common enough that it has a specialized function:  `.as_deref()`

```suggestion
        first.map(str::to_ascii_lowercase).as_deref()
        second.map(str::to_ascii_lowercase).as_deref()
```